### PR TITLE
TUI: add more keycodes follow-up PR

### PR DESCRIPTION
**Pull request by FractalWire**
Follow up on PR #9793

### CODE REVIEW

**Comment by FractalWire** [comment]
> keypad + doesn't work (sends ^[Ol).
TODO: We should at least log ignored things...

How should I do to create a new log entry ?

> in :terminal cat -v, keys 2 4 5 6 8 send escape sequences, whereas all other keys (0 1 3 7 9 + <Enter>. /*-) send the respective ASCII byte.
Maybe terminal.c needs a change.

Despite my comment in the previous PR, this is not something that we should 'fix'.
libvtermkey seems to implement a strict emulation of VT220 terminal. And according to [this](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-VT220-Style-Function-Keys), ASCII value are expected when not in application mode.
However, if you run nvim inside the terminal buffer, or if you switch to transmit mode (`tput smkx`), it will be recognized as `<kHome>` and friends.

This is different from what xterm/urxvt do (at least for me):
 * when not in transmit mode, xterm follows [this style](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-PC-Style-Function-Keys)
 * otherwise it follows the same style as [VT220](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-VT220-Style-Function-Keys)

In other word, if we want to have the function key working in nvim `:terminal`, we need to do what does xterm for example and switch style when the transmit mode changes.
This is libvterm responsability here.

File: runtime/doc/intro.txt [hunk 1]
```diff
@@ -373,8 +373,10 @@ notation	meaning		    equivalent	decimal value(s)	~
 <kMinus>	keypad -			*keypad-minus*
 <kMultiply>	keypad *			*keypad-multiply*
 <kDivide>	keypad /			*keypad-divide*
-<kEnter>	keypad Enter			*keypad-enter*
 <kPoint>	keypad Decimal point		*keypad-point*
+<kComma>	keypad comma key		*keypad-comma*
+<kEqual>	keypad equal key		*keypad-equal*
```
Comment (justinmk) [inline runtime/doc/intro.txt]: following the pattern from above, these would be:

```
<kComma>	keypad ,		*keypad-comma*
<kEqual>	keypad =		*keypad-equal*
```

**Comment by justinmk** [comment]
> How should I do to create a new log entry ?

See `log.h`, but the short answer is:

    ELOG("ignored input: '%s'", keys);

If this turns out to be noisy we can change it to `WLOG()` ("warning" level). But I'm interested to see how much it ends up logging.

> On a side note, I can't run make lint without it complaining about style in unrelated files. here is the output

That should be fixed now, the "ignore list" [was updated](https://github.com/neovim/bot-ci) and `make lint` will download  it automatically.

> not something that we should 'fix'.
> libvtermkey seems to implement a strict emulation of VT220 terminal. And according to [this](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-VT220-Style-Function-Keys), ASCII value are expected when not in transmit mode.

SGTM.

File: runtime/doc/intro.txt (hunk 1 above)
Comment (FractalWire) [reply runtime/doc/intro.txt]: :+1:

**Comment by FractalWire** [comment]
Here is a log output example with `<KP5>` being removed of the recognized keys in keymap.c : 

<details>
<summary>show/hide</summary>

```
ERROR 2019-03-29T11:37:30.643 1460  call_set_error:604: RPC: ch 1 sent an invalid message, closed.
ERROR 2019-03-30T13:57:06.066 3669  find_special_key:693: invalid special key name: '<args>'
ERROR 2019-03-30T13:57:06.066 3669  find_special_key:693: invalid special key name: '<args>'
ERROR 2019-03-30T13:57:06.068 3669  find_special_key:693: invalid special key name: '<bang>0 | set ft=man | else | call man#open_page(v:count, v:count1, <q-mods>, <f-args>) | endif'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<count>,<f-args>)<bar>call winrestview(s:svpos)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<line1>,<line2>call netrw#NetWrite(<f-args>)<bar>call winrestview(s:svpos)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<line2>call netrw#NetWrite(<f-args>)<bar>call winrestview(s:svpos)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<count>,0,0+<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<count>,1,0+<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<count>,1,2+<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<count>,1,4+<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<count>,0,6        ,<q-args>)'
ERROR 2019-03-30T13:57:06.069 3669  find_special_key:693: invalid special key name: '<count>,<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.070 3669  find_special_key:693: invalid special key name: '<bang>0,<q-args>)'
ERROR 2019-03-30T13:57:06.070 3669  find_special_key:693: invalid special key name: '<bang>0)'
ERROR 2019-03-30T13:57:06.070 3669  find_special_key:693: invalid special key name: '<cfile>')),netrw#CheckIfRemote())<cr>'
ERROR 2019-03-30T13:57:06.072 3669  find_special_key:693: invalid special key name: '<line1>, <line2>)'
ERROR 2019-03-30T13:57:06.072 3669  find_special_key:693: invalid special key name: '<line2>)'
ERROR 2019-03-30T13:57:10.649 3669  find_special_key:693: invalid special key name: '<KP5>'
ERROR 2019-03-30T13:57:11.578 3669  find_special_key:693: invalid special key name: '<KP5>'
ERROR 2019-03-30T13:57:11.802 3669  find_special_key:693: invalid special key name: '<KP5>'
```

</details>

> That should be fixed now, the "ignore list" was updated and make lint will download it automatically.

Didn't work for me, I still have the same errors. I'm not sure if there is an extra step to do apart from `make lint`. Should I build the bot-ci or something like that ?

**Comment by justinmk** [comment]
> Here is a log output example with <KP5> being removed of the recognized keys in keymap.c :

Pretty noisy, let's just leave it commented-out then. Probably somewhere in getchar.c (after it fails to resolve to a mapping) would be more useful, but not important for now.

> Should I build the bot-ci or something like that ?

No. Maybe try `rm -rf build`, or `make distclean` at the most drastic.

File: src/nvim/keymap.c [hunk 2]
```diff
@@ -275,6 +275,11 @@ static const struct key_name_entry {
   { K_K8,              "k8" },
   { K_K9,              "k9" },
 
+  { K_KCOMMA,          "kComma" },
+  { K_KCOMMA,          "KPComma" },
+  { K_KEQUAL,          "kEqual" },
+  { K_KEQUAL,          "KPEquals" },
```
Comment (justinmk) [inline src/nvim/keymap.c]: maybe group these next to `K_KPOINT` (line 265) ?

**Review by justinmk — APPROVED** [review]

**Comment by FractalWire** [comment]
> No. Maybe try rm -rf build, or make distclean at the most drastic.

Ah, yes, my output is different now, but I have [different errors](https://pastebin.com/raw/YPDMXq2x)

**Pull request closed without merging**